### PR TITLE
Weather cache#135

### LINKED ISSUE #135: Background worker for weather on dashboard
Implement background worker for weather on dashboard so that the page loads, and when the weather is loaded, that shows (but doesn't delay the whole page).

### PR DESCRIPTION
**Pull request by asmolentzov**
Introduces a scheduled Rake task to go through and get garden weather info every night.
- Adds weather_data column to gardens (jsonb data type)
- Updates Weather to take a garden rather than lat/long
- Updates views and garden/dashboard facades to give Weather a garden rather than lat/long
- Adds Rake task to update all gardens with weather data
- Added a scheduled task on Heroku
- Updates and adds tests as necessary
- Updates RainyDay: Changes from iterating over zip codes to iterating over gardens
- Updates to RainyDay spec and job spec so stubbing correctly with new Weather model.
- Updates .travis.yml to include Postgres version 9.6 which is necessary to use jsonb data type for new garden weather_data column
- Removes memoization in RainyDay model on class variable since it was hanging onto outdated values.
Closes #135

### CODE REVIEW

File: app/facades/dashboard_facade.rb [hunk 1]
```diff
@@ -3,11 +3,11 @@ def initialize(user)
     @user = user
   end
 
-  def weather(lat, long)
-    @_weather ||= Weather.new(lat, long)
+  def weather(garden)
+    @_weather ||= Weather.new(garden)
```
Comment (le3ah) [inline app/facades/dashboard_facade.rb]: Fancy!  I like this!

File: db/migrate/20190221011917_add_weather_data_to_gardens.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddWeatherDataToGardens < ActiveRecord::Migration[5.1]
+  def change
+    add_column :gardens, :weather_data, :jsonb
+  end
```
Comment (le3ah) [inline db/migrate/20190221011917_add_weather_data_to_gardens.rb]: I'm excited to learn more about the jsonb datatype!

**Comment by bendelonlee** [comment]
I didn't see anything in the error message about jsonb, did you?
But it is curious, why would rake db commands fail

**Comment by bendelonlee** [comment]
I found the jsonb error in a previous log

**Pull request merged**